src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,352 @@
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetStats(StatsInfoObjectCb.InfoCallback infoCb, IntPtr userData)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+
+            _getStatsObjectCb = infoCb;
+
+            _getStatsCb = (int result, IntPtr info, IntPtr userDataBack) =>
+            {
+                if(result != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "GetStats failed, Error - " + (StcError)result);
+                    StcErrorFactory.ThrowStcException(result);
+                }
+                return _getStatsObjectCb(new StcInfo(info), userDataBack);
+            };
+
+            int ret = Interop.Stc.Rule.GetStats(StcManagerImpl.Instance.GetSafeHandle(), _ruleHandle, _getStatsCb, userData);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "GetStats() failed , Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information of each application asynchronously. The callback is called for each application that used network in between timestamps specified.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <param name="userData">The user data passed to the callback function.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void ForeachStats(StatsInfoObjectCb.InfoCallback infoCb, IntPtr userData)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+
+            _foreachStatsObjectCb = infoCb;
+
+            _foreachStatsCb = (int result, IntPtr info, IntPtr userDataBack) =>
+            {
+                if(result != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "ForeachStats failed, Error - " + (StcError)result);
+                    StcErrorFactory.ThrowStcException(result);
+                }
+                return _foreachStatsObjectCb(new StcInfo(info), userDataBack);
```

Comment:
Please clarify ownership and lifetime of info

Reply:
I am not sure about it. But i think as 'new' is used, so there is no matter of scope of variable, and user have to call destroy for each 'info' object.

Reply:
`IntPtr info` is ONLY valid in callback scope. 
If you want to extand the lifetime of `info` you need to use some native API to copy